tests/test_linestring.py

```diff
@@ -208,5 +218,26 @@ def test_linestring_adapter_deprecated():
         asLineString(coords)
 
 
+def test_linestring_ctypes_deprecated():
+    line = LineString(((1.0, 2.0), (3.0, 4.0)))
+    with pytest.warns(ShapelyDeprecationWarning, match="ctypes"):
+        line.ctypes
+
+
+def test_linestring_array_interface_deprecated():
+    line = LineString(((1.0, 2.0), (3.0, 4.0)))
+    with pytest.warns(ShapelyDeprecationWarning, match="array_interface"):
+        line.array_interface()
+
+
+@unittest.skipIf(not numpy, 'Numpy required')
+def test_linestring_array_interface_numpy_deprecated():
+    import numpy as np
+
+    line = LineString(((1.0, 2.0), (3.0, 4.0)))
+    with pytest.warns(ShapelyDeprecationWarning, match="array interface"):
```

Comment:
Today I learned about the `match` keyword argument! I'm going to use this more often.